shell/platform/darwin/macos/framework/Source/FlutterPlatformNodeDelegateMac.mm

```diff
@@ -0,0 +1,104 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterPlatformNodeDelegateMac.h"
+
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterAppDelegate.h"
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterViewController.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterEngine_Internal.h"
+
+#include "flutter/shell/platform/common/cpp/accessibility_bridge.h"
+#include "flutter/third_party/accessibility/ax/ax_action_data.h"
+#include "flutter/third_party/accessibility/ax/ax_node_position.h"
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node.h"
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node_base.h"
+#include "flutter/third_party/accessibility/base/string_utils.h"
+#include "flutter/third_party/accessibility/gfx/geometry/rect_conversions.h"
+
+namespace flutter {  // namespace
+
+FlutterPlatformNodeDelegateMac::FlutterPlatformNodeDelegateMac(FlutterEngine* engine)
+    : engine_(engine) {
+  ax_platform_node_ = ui::AXPlatformNode::Create(this);
+  NSCAssert(ax_platform_node_, @"Failed to create platform node.");
+}
+
+FlutterPlatformNodeDelegateMac::~FlutterPlatformNodeDelegateMac() {
+  ax_platform_node_->Destroy();
+}
+
+gfx::NativeViewAccessible FlutterPlatformNodeDelegateMac::GetNativeViewAccessible() {
+  NSCAssert(ax_platform_node_, @"Platform node does not exist.");
+  return ax_platform_node_->GetNativeViewAccessible();
+}
+
+gfx::NativeViewAccessible FlutterPlatformNodeDelegateMac::GetParent() {
+  gfx::NativeViewAccessible parent = FlutterPlatformNodeDelegate::GetParent();
+  if (!parent) {
+    __strong FlutterEngine* strong_engine = engine_;
```

Comment:
I remember i read it somewhere, if I didn't retain the strong reference here, the engine_ may be dealloc right after this line if its owner release the reference

Reply:
That's true, but there are only two ways that can happen:
- It's released on another thread. This code isn't threadsafe, so if that happens something has already gone very, very wrong.
- A call you make in this block of code causes a chain of events that lead to deallocation as a side effect. Sometimes that's hard to reason about, and re-strongification can be useful as a default because of that, but it seems very clear that simple access of accessibility details can't possibly destroy the engine as a side-effect.